user guide: update internal files (#2556)
iterative/dvc.org: content/docs/user-guide/project-structure/internal-files.md

- `.dvc/tmp/index`: Directory for remote index files that are used for
  optimizing `dvc push`, `dvc pull`, `dvc fetch` and `dvc status -c` operations

- `.dvc/tmp/md5s`: This directory is used for optimization. It contains a SQLite

>   database, that stores hash values for files tracked in a DVC project, with
>   respective timestamps and inodes to avoid unnecessary file hash computations.

Reviewer: ```suggestion
  database that stores hash values for files tracked in a DVC project. It also saves
  the corresponding timestamps and inodes, to avoid unnecessary file hash
  computations.
```

> - `.dvc/tmp/links`: This directory contains a SQLite database, that stores a
>   list of links (from cache to <abbr>workspace</abbr>) created by DVC and is
>   used to cleanup your workspace when calling `dvc checkout`.

Reviewer: ```suggestion
- `.dvc/tmp/links`: This directory contains a SQLite database that stores a
  list of file links created by DVC (from cache to <abbr>workspace</abbr>).
  It's used to cleanup your workspace when calling `dvc checkout`.
```
